static/js/rename.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+window.onload = function() {
+  loadDeviceDetails();
+
+  if (hasGroupList()) {
+    var groups = getGroups();
+    var groupsSelect = document.getElementById("group-list")
+    for (var i in groups) {
+      var group = groups[i];
+      var groupOpt = document.createElement('option');
+      groupOpt.value = group;
+      groupOpt.innerHTML = group;
+      groupsSelect.appendChild(groupOpt);
+    }
+  }
+};
+
+function loadDeviceDetails() {
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.open("GET", "/api/device-info", true);
+  xmlHttp.setRequestHeader("Authorization", "Basic "+btoa("admin:feathers"))
+
+  xmlHttp.onload = async function() {
+    if (xmlHttp.status == 200) {
+      var response = JSON.parse(xmlHttp.response);
+      document.getElementById("new-name").placeholder=response.devicename;
```

Comment:
remember you've got jQuery available so you could just do things like `$("#new-name")...`